sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -0,0 +1,1255 @@
+            The ith index of the list holds the values of the ith layer.
+
+        deltas : list, length = n_layers - 1
+            The ith index of the list holds the difference between the
+            activations of the i + 1 layer and the backpropagated error.
+
+        coef_grad : list, length = n_layers - 1
+            The ith index contains the amount of change used to update the
+            coefficient parameters of the ith layer in an iteration.
+
+        intercept_grads : list, length = n_layers - 1
+            The ith index contains the amount of change used to update the
+            intercept parameters of the ith layer in an iteration.
+
+        Returns
+        -------
+        loss : float
+        grad : array-like, shape (number of nodes of all layers,)
+
+        """
+        self._unpack(packed_coef_inter)
+        loss, coef_grads, intercept_grads = self._backprop(X, y, activations,
+                                                           deltas, coef_grads,
+                                                           intercept_grads)
+        self.n_iter_ += 1
+        grad = _pack(coef_grads, intercept_grads)
+        return loss, grad
+
+    def _backprop(self, X, y, activations, deltas, coef_grads,
+                  intercept_grads):
+        """Compute the MLP loss function and its corresponding derivatives
+        with respect to each parameter: weights and bias vectors.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data.
+
+        y : array-like, shape (n_samples,)
+            The target values.
+
+        activations: list, length = n_layers - 1
+             The ith index of the list holds the values of the ith layer.
+
+        deltas : list, length = n_layers - 1
+            The ith index of the list holds the difference between the
+            activations of the i + 1 layer and the backpropagated error.
+            More specifically, deltas are gradients of loss with respect to z
+            in each layer, where z = wx + b is the value before passed through
+            activation function
+
+        coef_grad : list, length = n_layers - 1
+            The ith index contains the amount of change used to update the
+            coefficient parameters of the ith layer in an iteration.
+
+        intercept_grads : list, length = n_layers - 1
+            The ith index contains the amount of change used to update the
+            intercept parameters of the ith layer in an iteration.
+
+        Returns
+        -------
+        loss : float
+        """
+        n_samples = X.shape[0]
+
+        # Forward propagate
+        activations = self._forward_pass(activations)
+
+        # Get loss
+        loss = LOSS_FUNCTIONS[self.loss](y, activations[-1])
+        # Add L2 regularization term to loss
+        values = np.sum(np.array([np.sum(s ** 2) for s in self.coefs_]))
```

Comment:
OK so `alpha` is not per layer. That is OK.

But you should replace `np.sum(s ** 2)` by `np.dot(s.ravel(), s.ravel())`